src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -0,0 +1,1599 @@
+            tmp = Rotate(in tmp);
+            Sse.StoreScalar(dst + idx[2], tmp);
+            tmp = Rotate(in tmp);
+            Sse.StoreScalar(dst + idx[3], tmp);
+            tmp = GetHigh(in x);
+            Sse.StoreScalar(dst + idx[4], tmp);
+            tmp = Rotate(in tmp);
+            Sse.StoreScalar(dst + idx[5], tmp);
+            tmp = Rotate(in tmp);
+            Sse.StoreScalar(dst + idx[6], tmp);
+            tmp = Rotate(in tmp);
+            Sse.StoreScalar(dst + idx[7], tmp);
+        }
+
+        [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<float> VectorSum128(in Vector128<float> vector)
+        {
+            if (Sse3.IsSupported)
+            {
+                Vector128<float> partialSum = Sse3.HorizontalAdd(vector, vector);
+                return Sse3.HorizontalAdd(partialSum, partialSum);
+            }
+            else
+            {
+                Vector128<float> partialSum = Sse.Add(vector, Sse.MoveHighToLow(vector, vector));
+                // The control byte shuffles the four 32-bit floats of partialSum: ABCD -> BADC.
+                return Sse.Add(partialSum, Sse.Shuffle(partialSum, partialSum, 0xB1));
+            }
+        }
+
+        [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
+        private static Vector256<float> VectorSum256(in Vector256<float> vector)
+        {
+            Vector256<float> partialSum = Avx.HorizontalAdd(vector, vector);
+            return Avx.HorizontalAdd(partialSum, partialSum);
+        }
+
+        [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<float> VectorMax128(in Vector128<float> vector)
+        {
+            Vector128<float> x1 = Sse.Shuffle(vector, vector, 0xB1);
+            Vector128<float> partialMax = Sse.Max(vector, x1);
+            x1 = Sse.Shuffle(partialMax, partialMax, 0x02);
+            return Sse.MaxScalar(partialMax, x1);
+        }
+
+        [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
+        private static Vector256<float> VectorMax256(in Vector256<float> vector)
+        {
+            Vector256<float> x1 = Avx.Shuffle(vector, vector, 0xB1);
+            Vector256<float> partialMax = Avx.Max(vector, x1);
+            x1 = Avx.Shuffle(partialMax, partialMax, 0x02);
+            return Avx.Max(partialMax, x1);
+        }
+
+        [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<float> GetAbsMask128()
+        {
+            return Sse2.IsSupported ?
+                Sse.StaticCast<int, float>(Sse2.SetAllVector128(0x7FFFFFFF)) :
+                Sse.SetAllVector128(BitConverter.Int32BitsToSingle(0x7FFFFFFF));
+        }
+
+        [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
+        private static Vector256<float> GetAbsMask256()
+        {
+            return Avx.StaticCast<int, float>(Avx.SetAllVector256(0x7FFFFFFF));
```

Comment:
To clarify, do you mean that there should be a `static readonly` variable like `Vector256<float> absMask256 =  Avx.StaticCast<int, float>(Avx.SetAllVector256(0x7FFFFFFF));` instead of a function `GetAbsMask256`?